app/addListing/ctrl.js

```diff
@@ -0,0 +1,22 @@
+function ctrl($scope, $routeParams, service) {
```

Comment:
Try to avoid using $scope.  In the template of a component, you can use `$ctrl` to access models.